ui/job-view/details/PinBoard.jsx

```diff
@@ -592,10 +607,20 @@ class PinBoard extends React.Component {
                 >
                   <Button
                     className={`${!isLoggedIn ? 'disabled' : ''} dropdown-item`}
-                    onClick={() => !isLoggedIn || this.retriggerAllPinnedJobs()}
+                    onClick={evt =>
+                      !isLoggedIn || this.retriggerAllPinnedJobs(evt.shiftKey)
+                    }
                   >
                     Retrigger all
                   </Button>
+                  {this.state.timesModal && (
+                    <RetriggerMultiplePrompt
+                      isOpen={this.state.timesModal}
```

Comment:
Instead of referencing ``this.state.foo`` please use destructuring above.  So in line 391, you would just add ``timesModal`` and ``retriggerTimesJobs``.  Please also do the same for anything from ``this.props``.

    const {